src/protocol/payload/mod.rs

```diff
@@ -126,10 +126,14 @@ impl VarStr {
     }
 }
 
-#[derive(Debug, PartialEq)]
-struct Hash([u8; 32]);
+#[derive(Debug, PartialEq, Clone)]
```

Comment:
Is `Clone` here necessary?

Reply:
Ah its not currently. I had use for it temporarily, will remove.